examples/platforms/nrf52840/alarm.c

```diff
@@ -563,3 +571,20 @@ void RTC_IRQ_HANDLER(void)
         }
     }
 }
+
+#if OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
+void otPlatTimeSet(uint64_t aTime)
+{
+    sNetworkTimeOffset = (int64_t)aTime - (int64_t)nrf5AlarmGetCurrentTime();
```

Comment:
If the platform layer is only maintaining the offset, how about moving this from the platform layer into OT core?  We should try to keep the platform layer as simple as possible.

Reply:
I think it is fine to have the platform layer update the IE.  We could simply pass the network time offset down as part of the `otRadioFrame` struct.  If we ever support a capability where platform can provide the AES function (some radios support this as part of their transmit/receive operations), then having the platform layer update the IE would be easier to support.  Thoughts?

Reply:
Ok, it does make sense. I have updated the PR to maintain offset in OT core, and still update IE in platform layer.